cni-plugin/deployment/linkerd-cni.yaml

```diff
@@ -0,0 +1,181 @@
+  inbound_ports_to_ignore: "[4190,4191]"
+  outbound_ports_to_ignore: ""
+  simulate: "false"
+  log_level: "debug"
+  # The CNI network configuration to install on each node. The special
+  # values in this config will be automatically populated.
+  cni_network_config: |-
+    {
+      "name": "linkerd-cni",
+      "type": "linkerd-cni",
+      "log_level": "__LOG_LEVEL__",
+      "policy": {
+          "type": "k8s",
+          "k8s_api_root": "https://__KUBERNETES_SERVICE_HOST__:__KUBERNETES_SERVICE_PORT__",
+          "k8s_auth_token": "__SERVICEACCOUNT_TOKEN__"
+      },
+      "kubernetes": {
+          "kubeconfig": "__KUBECONFIG_FILEPATH__"
+      },
+      "linkerd": {
+        "incoming-proxy-port": __INCOMING_PROXY_PORT__,
+        "outgoing-proxy-port": __OUTGOING_PROXY_PORT__,
+        "proxy-uid": __PROXY_UID__,
+        "ports-to-redirect": __PORTS_TO_REDIRECT__,
+        "inbound-ports-to-ignore": __INBOUND_PORTS_TO_IGNORE__,
+        "outbound-ports-to-ignore": __OUTBOUND_PORTS_TO_IGNORE__,
+        "simulate": __SIMULATE__
+      }
+    }
+---
+# This manifest installs the linkerd CNI plugins and network config on
+# each master and worker node in a Kubernetes cluster.
+kind: DaemonSet
+apiVersion: extensions/v1beta1
+metadata:
+  name: linkerd-cni
+  namespace: linkerd
+  labels:
+    k8s-app: linkerd-cni
+spec:
+  selector:
+    matchLabels:
+      k8s-app: linkerd-cni
+  updateStrategy:
+    type: RollingUpdate
+    rollingUpdate:
+      maxUnavailable: 1
+  template:
+    metadata:
+      labels:
+        k8s-app: linkerd-cni
+      annotations:
+        # This, along with the CriticalAddonsOnly toleration below,
+        # marks the pod as a critical add-on, ensuring it gets
+        # priority scheduling and that its resources are reserved
+        # if it ever gets evicted.
+        scheduler.alpha.kubernetes.io/critical-pod: ''
+    spec:
+      nodeSelector:
+        beta.kubernetes.io/os: linux
+      hostNetwork: true
+      tolerations:
+        # Make sure linkerd-cni gets scheduled on all nodes.
+        - effect: NoSchedule
+          operator: Exists
+        # Mark the pod as a critical add-on for rescheduling.
+        - key: CriticalAddonsOnly
+          operator: Exists
+        - effect: NoExecute
+          operator: Exists
+      serviceAccountName: linkerd-cni
+      # Minimize downtime during a rolling upgrade or deletion; tell Kubernetes to do a "force
+      # deletion": https://kubernetes.io/docs/concepts/workloads/pods/pod/#termination-of-pods.
+      terminationGracePeriodSeconds: 5
+      containers:
+        # This container installs the linkerd CNI binaries
+        # and CNI network config file on each node. The install
+        # script copies the files into place and then sleeps so
+        # that Kubernetes doesn't keep trying to restart it.
+        - name: install-cni
+          image: gcr.io/linkerd-io/cni-plugin:dev-bd117b06-x27s
+          command: ["/install-cni.sh"]
```

Comment:
You can remove this field and rely on the default ENTRYPOINT from the container. That will be more future-proof if the ENTRYPOINT ever changes (as it does in my suggested refactor to your Dockerfile).